codec/test_codec.go

```diff
@@ -887,3 +890,90 @@ func TestMultipleTags(codec GeneralCodec, t testing.TB) {
 		require.Empty(output.NoTags)
 	}
 }
+
+func TestMap(codec GeneralCodec, t testing.TB) {
```

Comment:
Maybe we should have all these map tests under one `TestMap` function with each of these tests as a subtest.

We should also test the following cases:

* Map with struct keys/values
* Map with interface values
* Map with slice values
* Map with array keys/values
* Test to ensure encoding is canonical

Reply:
I added more tests in 44a50fbf4126b84e41d6d306e7f6e25505c594cc

Reply:
Can we also add a case for map of maps? i.e `map[int32]map[int32]MyInnerStruct3`